continuousprint/__init__.py

```diff
@@ -186,7 +191,31 @@ def cancel_print(self):
         self._msg("Print cancelled", type="error")
         self._printer.cancel_print()
 
+    def bed_cooldown(self):
+        self._logger.info("Running bed cooldown script")
+        bed_cooldown_script = self._settings.get(["cp_bed_cooldown_script"]).split("\n")
+        self._printer.commands(bed_cooldown_script, force=True)
+        self._logger.info("Preparing for Bed Cooldown")
+        self._printer.set_temperature("bed", 0)  # turn bed off
+        start_time = time.time()
+
+        while (time.time() - start_time) <= (60 * float(self._settings.get(["bed_cooldown_timeout"]))):  # timeout converted to seconds
```

Comment:
Yes! Let's talk about this, 
I understand this solution I wrote up is pretty computationally wasteful.
I put together a solution using util.RepeatedTimer like you suggested and in the implementation, I put together I was facing several issues. 
When calling util.RepeatedTimer it wouldn't halt the continuous print flow so it would run like this:

PRINT FINISHES -> COOLDOWN TIMER LOOP STARTS-> BED CLEAR SCRIPT (prematurely) RUNS -> COOLDOWN LOOP IS STILL RUNNING -> COOLDOWN LOOP COMPLETES -> NEXT PRINT STARTS

I recognize there is a  high probability that I was doing something wrong but I spent a bit of time working on this egg and I couldn't crack it 🥚. If you would like I could take another crack at it. Let me know what you think

Reply:
@smartin015

Reply:
Sounds like switching to a timer solution would need a lot more work on the driver side. Since this is an opt-in feature, let's run with your solution for now so we can get a fix in place, and optimize later if it becomes a problem.